docs/proposals/shoot-control-plane-migration/README.md

```diff
@@ -0,0 +1,234 @@
+* [Etcd Backup Restore](https://github.com/gardener/etcd-backup-restore) already backs up the shoot etcd which is running in the shoot namespace of the source seed cluster. We can re-use this completely while restoring the shoot etcd on the destination seed cluster.
+
+## Proposed Solution
+
+### Backup and Restoration
+
+The backup and restoration of the following resources that are currently not stored in the shoot apiserver/etcd is not covered by the [reuse](#reuse) mentioned above.
+
+* Resources from the shoot namespace in the seed.
+  * Terraform resources such as configmaps and secrets.
+  * Machine Controller Manager resources such as machinedeployments, machinesets, machines, machineclasses, secrets etc.
+* Resources from the shoot backup namespace in the seed.
+  * Terraform resources such as configmaps and secrets.
+
+All the additional resources listed above will be backed up in the same object store as the what is used for backing up the shoot etcd.
+
+* The backup will be watch-based and not schedule based.
+* The backup will be full backups at longer intervals followed by incremental backups at smaller intervals similar to the etcd continuous/incremental backup.
+* While taking backup it will take care of dependencies. This is needed since while restoring from it, validation checks needs to be passed.
+    
+    * But the granularity of the changes stored at the incremental backups would be at the object level and not the actual field-level changes. This is different from the etcd continuous/incremental backups where the incremental changes are recorded at the field level.
+
+#### Components of the solution
+
+Taking the idea from the framework setup for etcd-backup-restore, we will introduce new component named 
+**shoot-backup-restore** per shoot in seed namespace. This will backup the control plane component of shoot only from seed namespace.
+
+##### shoot-backup-restore implementation
+
+* It will get list of objects to be backed up via configmap shoot-resources.
+
+  * TODO: Come up json format for dependencies configmap
+  * Initial thought (not full proof) is To have nested dependency fields list  
+  * Example configmap content:
+  ```yaml
+  awsmachineclass: spec.sercretRef
+  machines: spec.class
+  machineset: ...
+  machinedeployment: ...
+  configmap: ...
+  secret: ...
+  roles: ...       # May not be required?
+  rolebinding: ... # May not be Required?
+  service: ...     # May not be Required?
+
+  ```
+
+  * TODO: Give thought about OwnerReferences
+  * TODO: Think on dependency hooks
+
+* It will apply watch on seed's apiserver for those objects and takes backup of full object including its dependency.
+
+* It will have an `/initialize` endpoint to initialize cluster.
+  * This endpoint will be called by gardener in each reconcile.
+  * As a part of endpoint, it should try to restore the objects from backups stored in bucket only if resource revision is older than last backup.
```

Comment:
Why bucket? Why not use a K8S cluster for backup?

Reply:
Which K8s cluster?  You wan't maintain new HA backup cluster? Or are you thinking of replicating seed in gardener?

Reply:
I vote for object store backup. It will be there anyway for the shoot etcd backup. It is not as if we are introducing something new.